### PR TITLE
Resolve memory allocation issue + minor updates

### PR DESCRIPTION
**Pull request by nrasadi**
As depicted in the original notebook, the user requires at least 28GB free memory space, which may not be available in many cases.   
This PR resolves this issue.

Moreover, it comes with minor updates including the way of loading and saving files and showing examples.

### CODE REVIEW

**Comment by calebrob6** [comment]
Hi @nrasadi, thanks for the contribution!

It looks like there is an issue with the labels, the example images at the end of the notebook do not match up with their labels:
image

can you investigate this? 

(just a quick guess, `x_val = np.load(str(path_imagenet_val_dataset/"x_val_1.npy"))` may should be `x_val = np.load(str(path_imagenet_val_dataset/"x_val_0.npy"))`?)

**Comment by calebrob6** [comment]
Also, this change will break the second notebook -- it expects `x_val.npy` to exist. Do you mind changing to second notebook to run on these batched samples and verify that you get the same accuracy results?

**Comment by nrasadi** [comment]
> Hi @nrasadi, thanks for the contribution!
> 
> It looks like there is an issue with the labels, the example images at the end of the notebook do not match up with their labels:
> image
> 
> can you investigate this?
> 
> (just a quick guess, `x_val = np.load(str(path_imagenet_val_dataset/"x_val_1.npy"))` may should be `x_val = np.load(str(path_imagenet_val_dataset/"x_val_0.npy"))`?)

Oh! Sorry. There was a little mistake. I will take care of it.

**Comment by nrasadi** [comment]
> Also, this change will break the second notebook -- it expects `x_val.npy` to exist. Do you mind changing to second notebook to run on these batched samples and verify that you get the same accuracy results?

Sure. I will fix this.

**Comment by calebrob6** [comment]
It looks like the validation accuracies do not match up now -- my original notebook has top-1 of 0.71256 while yours has 0.61778.

**Comment by nrasadi** [comment]
I  updated the second notebook to be compatible with the first one. However, I don't get the same results.
|   | Reported Accuracy (%) | Obtained Results |
| - | ------------------------------- | -------------------- |
| Top-1 | 71.3   | 61.8   |
| Top-5 | 90.0   | 84      |

**Comment by calebrob6** [comment]
Right, can you look into this? I think they should match up

**Comment by nrasadi** [comment]
> It looks like the validation accuracies do not match up now -- my original notebook has top-1 of 0.71256 while yours has 0.61778.

Yes. You're right. But I don't know why!

**Comment by nrasadi** [comment]
> Right, can you look into this? I think they should match up

Yes, of course. But I think the results should be the same. Could it be due to the difference between the Keras versions?

**Comment by calebrob6** [comment]
I'd be surprised as this is a large difference. Are images in the correct order with the labels? Can you confirm that the preprocessing is working in the same way?

**Comment by nrasadi** [comment]
Could you please have a look at [this issue](https://github.com/keras-team/keras/issues/8672)?

**Comment by nrasadi** [comment]
> I'd be surprised as this is a large difference. Are images in the correct order with the labels? Can you confirm that the preprocessing is working in the same way?

I double checked the notebooks. They don't seem different than the original ones.

**Comment by calebrob6** [comment]
Something has to be different if the results are different

**Comment by nrasadi** [comment]
I'm gonna give up. As the last try, can you find the Keras version you were used to get those results?   
Just to make sure nothing is changed since then.

**Comment by calebrob6** [comment]
> As the last try, can you find the Keras version you were used to get those results?

Good question, I'll try to reproduce my results now.

**Comment by calebrob6** [comment]
@nrasadi, I'm able to reproduce the results as is with tensorflow=2.3.1 and keras=2.4.3

**Comment by nrasadi** [comment]
> @nrasadi, I'm able to reproduce the results as is with tensorflow=2.3.1 and keras=2.4.3

Thank you very much.

**Comment by nrasadi** [comment]
I was able to reproduce the results for VGG19 model with Tensorflow==2.3.0.
Top-1 =  71.248 %
Top-5 =  89.986 %

**Comment by nrasadi** [comment]
I will gradually check the other available models in the next few days if it would help.

**Comment by nrasadi** [comment]
Just in case, I've added a tensor-based top_k_accuracy calculation but I mistakenly haven't created a seperate PR for that.

**Comment by calebrob6** [comment]
Thanks, great stuff @nrasadi ! Just a few changes if you don't mind and I'll be happy to merge!

For `1. Preprocess ImageNet validation set.ipynb`:
- Can you remove tensorflow imports (just to clean it up -- tensorflow isn't used here)
- Can you hardcode the paths to `ILSVRC2012_validation_ground_truth.txt`, `meta.mat`, ... to the `data/` directory
- `ut.humansize` is not defined

For `2. Benchmark Keras pretrained models on ImageNet`:
- The % complete messages are not behaving correctly.

**Comment by nrasadi** [comment]
> Just a few changes if you don't mind and I'll be happy to merge!

I fixed the mentioned issues. Please review them.
Again, thanks for your really helpful notebooks and spending your precious time on this PR. 🙏

**Comment by calebrob6** [comment]
Awesome, thanks for the contribution! This repo will definitely be more useful to others now.

Best,
Caleb
